### PR TITLE
feat: update to fastify 4

### PR DESCRIPTION
**Pull request by danielmitrov**
## In this PR:

- Updated fastify's major

### Checklist:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/ducktors/turborepo-remote-cache/pulls) for the same update/change?

* [x] Have you lint your code with `pnpm lint` locally prior to submission?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran build with `pnpm build` of your changes locally?
* [x] Have you successfully ran tests with `pnpm test` of your changes locally?
* [x] Have you commit using [Conventional Commits](https://github.com/ducktors/turborepo-remote-cache#how-to-commit)?

### CODE REVIEW

**Comment by danielmitrov** [comment]
I believe this will solve #6.
Still need to check it later at work, since we are having this issue in our cache server.

@fox1t

**Comment by fox1t** [comment]
Good catch. Since we are not introducing breaking changes, this should be safe.

File: package.json [hunk 1]
```diff
@@ -35,17 +35,16 @@
...
+    "@sinclair/typebox": "0.25.21",
     "@vercel/node": "1.12.1",
     "ajv": "8.11.0",
     "aws-sdk": "2.1053.0",
     "azure-storage": "^2.10.7",
     "boom": "7.3.0",
     "close-with-grace": "1.1.0",
     "env-schema": "5.1.0",
-    "fastify": "3.29.4",
-    "fastify-autoload": "3.9.0",
```
Comment (fox1t) [inline package.json:47]: I suppose we are not using autoload. Good catch. It should be a leftover.

File: src/plugins/remote-cache/routes/get-artifact.ts [hunk 2]
```diff
@@ -24,7 +25,7 @@ export const getArtifact: RouteOptions<
 
     try {
       const artifact = await this.location.getCachedArtifact(artifactId, teamId)
-      reply.send(artifact)
+      return reply.send(artifact)
```
Comment (fox1t) [inline src/plugins/remote-cache/routes/get-artifact.ts:28]: Do not return and use reply send together.

File: src/plugins/remote-cache/routes/schema.ts [hunk 3]
```diff
@@ -5,15 +5,15 @@ const querystring = Type.Object(
     teamId: Type.Optional(Type.String()),
     slug: Type.Optional(Type.String()),
   },
-  { additionalParameters: false },
+  { additionalProperties: false },
```
Comment (fox1t) [inline src/plugins/remote-cache/routes/schema.ts:8]: Another good catch!

**Review by fox1t — CHANGES_REQUESTED** [review]

File: src/plugins/remote-cache/routes/get-artifact.ts (hunk 2 above)
Comment (danielmitrov) [reply src/plugins/remote-cache/routes/get-artifact.ts:28]: According to fastify's docs:
https://www.fastify.io/docs/latest/Reference/Reply/#streams
> When using async-await you will need to return or await the reply object

Comment (fox1t) [reply src/plugins/remote-cache/routes/get-artifact.ts:28]: We should then also add that specific header. This must be changed from 3 to 4. Thanks for pointing this out.

**Comment by fox1t** [comment]
Could you address the only comment that remains and resolve the conflicts?

**Comment by danielmitrov** [comment]
> Could you address the only comment that remains and resolve the conflicts?

Done 😊

**Comment by itayganor** [comment]
Would love to see this merged! 🎉

**Review by fox1t — DISMISSED** [review]
LGTM

**Comment by fox1t** [comment]
This is a minor bump @matteovivona

**Comment by fox1t** [comment]
@danielmitrov I am not sure about the "This branch has conflicts that must be resolved" message. Can you look into it so we can merge and publish this PR?

**Comment by fox1t** [comment]
Ok, it is about the package.json and pnpm-lock.

**Comment by danielmitrov** [comment]
Yeah, sure. Fixed the conflicts!

**Comment by danielmitrov** [comment]
@matteovivona can you approve this PR so I can merge it? Or do you want @fox1t to approve it?

**Review by matteovivona — APPROVED** [review]

**Pull request merged**